src/bundle.ts

```diff
@@ -33,32 +30,24 @@ export class Bundle {
 	// The sourcemap must be synchronously available for formatError.
 	sourcemap = {} as RawSourceMap;
 
-	constructor(file: string, log: Log, config: esbuild.BuildOptions) {
-		this.file = file;
-		this.log = log;
-
-		this.config = {
-			target: "es2015",
-			...config,
-			entryPoints: [file],
-			sourcemap: true,
-			bundle: true,
-			write: false,
-			incremental: true,
-			platform: "browser",
-			define: {
-				"process.env.NODE_ENV": JSON.stringify(
-					process.env.NODE_ENV || "development",
-				),
-				...config.define,
-			},
-		};
+	constructor(
+		private file: string,
+		private log: Log,
+		private config: esbuild.BuildOptions,
+		private emitter: EventEmitter,
+	) {
+		this.config = { ...config, entryPoints: [file] };
 	}
 
 	dirty() {
 		if (this._dirty) return;
 		this._dirty = true;
 		this.deferred = new Deferred();
+		if (this.buildInProgress) this.write();
```

Comment:
I don't remember why this was necessary anymore. Can you delete and see if one of the tests fails?

It seems the rewriting of `this.deferred` above will ensure any waiters on the old promise will resolve only once the new build happens, and that should happen with the next normal write.

Reply:
Good call this can be removed. All tests are passing without this line 👍